.pre-commit-config.yaml

```diff
@@ -0,0 +1,59 @@
+---
+default_stages: [pre-commit]
+
+default_install_hook_types: [pre-commit, commit-msg]
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: mixed-line-ending
+
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.32.0
+    hooks:
+      - id: yamllint
+        args: [-c=.yamllint]
+
+  - repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.36.0
+    hooks:
+      - id: markdownlint
+      - id: markdownlint-fix
+
+  - repo: https://github.com/talos-systems/conform
+    rev: v0.1.0-alpha.27
+    hooks:
+      - id: conform
+        stages:
+          - commit-msg
+
+  - repo: https://github.com/koalaman/shellcheck-precommit
+    rev: v0.10.0
+    hooks:
+      - id: shellcheck
+        args: ["--severity=warning"]
+
+  - repo: https://github.com/priv-kweihmann/oelint-adv
+    rev: 5.3.2
+    hooks:
+      - id: oelint-adv
+        args: [--rulefile=.oelint-ruleset.json, --noinfo, --nowarn, --quiet]
```

Comment:
I also based this on https://github.com/3mdeb/meta-trenchboot/pull/28. I just removed the info/warning suppression and I noticed that there is a warning like this:
```
On a multiline assignment, line indent is desirable. 2 set, 4 desirable
```

This seemed like something that we want, because we usually want to add 4 spaces of indent in multiline assignments. However, the warning is quite different:

> For better readability the [Yocto project style guide](https://docs.yoctoproject.org/contributor-guide/recipe-style-guide.html#variable-formatting) suggests to use a common indent (minimum 4 spaces) matching the first " on the starting line

This is different from our style, where we use 4 indents. It makes some assignments look quite awkward imo:
```
EXTRA_OEMAKE = '\
                DESTDIR="${D}" \
                PREFIX="${prefix}" \
'
```

`oelint` allows creating cusotm rules in Python. We can try to create one that always wants a 4 space indent in multiline assignments. wdyt?

Reply:
Aside from there are many different warnings/info that I think are a little overkill such as:

```
warning:oelint.vars.dependsordered:'RDEPENDS' entries should be ordered alphabetically
```

Below is the full log
[oelint_info.txt](https://github.com/user-attachments/files/15785725/oelint_info.txt)

Reply:
@PLangowski, could such check be then excluded from `.oelint-ruleset.json`?

Reply:
This has been resolved in the meta-dts PR.

Reply:
@macpijan, this matter have been settled, please resolve this thread if you are satisfied.

Reply:
This thread is outdated, closing.